### ISSUE TITLE
[clang-tidy] false positive when using default comparisons

### ISSUE THREAD

**Issue by ollef** [llvm/llvm-project#59124]
Consider the following program:

`test.cpp`:
```c++
#include <compare>

struct Test {
  int X;
  friend auto operator<=>(const Test& LHS, const Test& RHS) = default;
};

bool test() {
  return Test{1} < Test{2};
}
```

and clang-tidy config:

`config`:
```yaml
Checks: 'readability-identifier-naming'
CheckOptions:
  - key:             readability-identifier-naming.VariableCase
    value:           CamelCase
```

Running

```shell
clang-tidy test.cpp -config-file=config -- -std=c++20
```

results in the following warning:

```console
test.cpp:5:69: warning: invalid case style for variable 'cmp' [readability-identifier-naming]
  friend auto operator<=>(const Test& LHS, const Test& RHS) = default;
                                                                    ^
                                                                    Cmp
```

I assume that this comes from the code generated for the default implementation, but I would not expect warnings in that code.

This happens when using clang-tidy 15.0.1 built from source:

```console
% clang-tidy --version
LLVM (http://llvm.org/):
  LLVM version 15.0.1
  Optimized build with assertions.
  Default target: x86_64-unknown-linux-gnu
  Host CPU: znver2
```

**Comment by PiotrZSL** [comment]
Fixed by 1c282052624f9d0bd273bde0b47b30c96699c6c7
